Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -114,22 +114,31 @@ extension BrowserViewController: TopToolbarDelegate {
         if !handleIPFSSchemeURL(url, visitType: .unknown) {
           tabManager.selectedTab?.reload()
         }
-      } else if !isPrivateMode, url.domainURL.schemelessAbsoluteDisplayString.endsWithSupportedSNSExtension, let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode) {
-        Task { @MainActor in
-          let currentStatus = await rpcService.snsResolveMethod()
-          switch currentStatus {
-          case .ask:
-            // show name service interstitial page
-            showSNSDomainInterstitialPage(originalURL: url, visitType: .unknown)
-          case .enabled:
-            if let resolvedURL = await resolveSNSHost(url.schemelessAbsoluteDisplayString, rpcService: rpcService) {
+      } else if !isPrivateMode,
+                let url = topToolbar.currentURL,
+                DecentralizedDNSHelper.isSupported(domain: url.domainURL.schemelessAbsoluteDisplayString),
+                let rpcService = BraveWallet.JsonRpcServiceFactory.get(privateMode: isPrivateMode),
+                let decentralizedDNSHelper = DecentralizedDNSHelper(
```

Comment:
Do you think it might be more feasible to convert the operation for 

```
DecentralizedDNSHelper.isSupported
let rpcService =  BraveWallet.JsonRpcServiceFactory
let decentralizedDNSHelper = DecentralizedDNSHelper
```
into a function and return a tuple for service and helper and use it. This logic is hard to read inside clause and it is repeated in many places as far as I can see.

Reply:
Updated in [68c55ce](https://github.com/brave/brave-ios/pull/7153/commits/68c55ceba94c80a1c596ab235634a317b1093553) 🙂